src/all/hitomi/src/eu/kanade/tachiyomi/extension/all/hitomi/Hitomi.kt

```diff
@@ -657,11 +685,53 @@ class Hitomi(
         return hash.replace(Regex("""^.*(..)(.)$"""), "$2/$1")
     }
 
+    override fun setupPreferenceScreen(screen: PreferenceScreen) {
+        ListPreference(screen.context).apply {
+            key = PREF_IMAGETYPE
+            title = "Images Type"
+            entries = arrayOf("webp", "avif", "jxl")
+            entryValues = arrayOf("webp", "avif", "jxl")
+            summary = "Clear chapter cache to apply changes"
+            setDefaultValue("webp")
+        }.also(screen::addPreference)
+    }
+
+    private fun List<Int>.toBytesList(): ByteArray = this.map { it.toByte() }.toByteArray()
+    private val signatureOne = listOf(0xFF, 0x0A).toBytesList()
+    private val signatureTwo = listOf(0x00, 0x00, 0x00, 0x0C, 0x4A, 0x58, 0x4C, 0x20, 0x0D, 0x0A, 0x87, 0x0A).toBytesList()
+    fun ByteArray.startsWith(byteArray: ByteArray): Boolean {
+        if (this.size < byteArray.size) return false
+        return this.sliceArray(byteArray.indices).contentEquals(byteArray)
+    }
+
+    private fun Intercept(chain: Interceptor.Chain): Response {
+        val response = chain.proceed(chain.request())
+        if (response.headers["Content-Type"] != "application/octet-stream") {
+            return response
+        }
+
+        val bytesArray = response.body.bytes()
```

Comment:
this will consume the response body so returning it as response later cause exception

Reply:
I don't understand, can you please explain more? Thank you.
Sorry if this is a weird question 🙏

Reply:
calling `response.body.*` methods like `.string()` or `.bytes()` etc, "consume" the body content and close the reponse, so it no longer will have the body contents. 
https://square.github.io/okhttp/5.x/okhttp/okhttp3/-response/index.html#1708455149%2FProperties%2F1481842020

If you want to check the response without closing it, [peek into it.](https://github.com/keiyoushi/extensions-source/blob/d7dc91fd8b9b593c1e49a5a4a832991b8813f79a/src/en/luascans/src/eu/kanade/tachiyomi/extension/en/luascans/LuaScans.kt#L26)

Reply:
Thanks for explaining it to me :pray: 
Sorry, but I think the commit will be pushed a bit late because I'm currently on my phone.